paperqa/contrib/openreview_paper_helper.py

```diff
@@ -0,0 +1,163 @@
+        """Creates a string containing the id, title, and abstract of all submissions."""
+        submission_info_string = ""
+        for submission in submissions:
+            paper = {
+                "submission_id": submission.id,
+                "title": submission.content["title"]["value"],
+                "abstract": submission.content["abstract"]["value"],
+            }
+            submission_info_string += f"{paper}\n"
+        return submission_info_string
+
+    async def fetch_relevant_papers(self, question: str) -> dict[str, Any]:
+        """Get relevant papers for a given question using LLM."""
+        submissions = self.get_submissions()
+        submission_string = self.create_submission_string(submissions)
+
+        if len(submission_string) > self.settings.parsing.chunk_size:
+            chunks = [
+                submission_string[i : i + self.settings.parsing.chunk_size]
+                for i in range(
+                    0, len(submission_string), self.settings.parsing.chunk_size
+                )
+            ]
+        else:
+            chunks = [submission_string]
+        relevant_papers = []
+        for chunk in chunks:
+            logger.info(f"Fetching relevant papers for question: {question}")
+            relevant_papers += await self._get_relevant_papers_chunk(question, chunk)
+        subs = [s for s in submissions if s.id in set(relevant_papers)]
+        await self.download_papers(subs)
+        return {sub.id: sub for sub in subs}
+
+    async def _get_relevant_papers_chunk(self, question: str, chunk: str) -> list[Any]:
+        prompt = (
+            chunk
+            + "You are the helper model that aims to get up to 20 most relevant papers for the user's question. "
+            + "User's question:\n"
+        )
+
+        response = await self.llm_model.call_single(
+            messages=[Message(role="user", content=prompt + question)],
+            output_type=RELEVANT_PAPERS_SCHEMA,
+        )
+
+        content = json.loads(str(response.text))
+        return [p["submission_id"] for p in content["suggested_papers"]]
+
+    async def download_papers(self, submissions: list[Any]) -> None:
+        """Download PDFs for given submissions."""
+        downloaded_papers = Path(self.settings.paper_directory).rglob("*.pdf")
+        downloaded_ids = [p.stem for p in downloaded_papers]
+        logger.info("Downloading PDFs for relevant papers.")
+        for submission in submissions:
+            if submission.id not in downloaded_ids:
+                await self._download_pdf(submission)
+
+    async def _download_pdf(self, submission: Any) -> bool:
+        """Download a single PDF."""
+        pdf_link = f"https://openreview.net/{submission.content['pdf']['value']}"
+        async with httpx.AsyncClient() as client:
+            response = await client.get(pdf_link)
+        SUCCESS_CODE = 200
+        if response.status_code == SUCCESS_CODE:
```

Comment:
```suggestion
        if response.status_code == httpx.codes.OK.value:
```

We don't need to redefine status codes in our code